pandas/tests/frame/methods/test_reindex.py

```diff
@@ -113,6 +116,11 @@ def test_reindex_date_fill_value(self):
         )
         tm.assert_frame_equal(res, expected)
 
+        # only reindexing rows
+        with tm.assert_produces_warning(FutureWarning):
```

Comment:
why is this a warn?

Reply:
xref https://github.com/pandas-dev/pandas/pull/44798#issuecomment-988264205.

@jorisvandenbossche maybe open dedicated issue for discussion/visibility.

Reply:
I think this PR can serve that purpose?

Reply:
could use a `match=...` to clarify